build.sbt

```diff
@@ -95,16 +95,28 @@ lazy val jacksonParameterNamesJavacSettings = Seq(
 )
 
 // configuration of sbteclipse
+// Despite being from the Eclipse plugin these settings impact all the SBT project. As a consequence these default
+// values prevent you from using scala code in this sample app. Read the comments marked as 'SCALA SUPPORT'
+// to have sbt compile and use both java and scala source files.
+// DISCLAIMER: Enabling scala support has the side effect, in Eclipse IDE, to add a lot of clutter on the Project View.
 lazy val eclipseSettings = Seq(
+
+  // SCALA SUPPORT 1: Comment the (Java flavor) and uncomment the (ScalaIDE flavor)
   EclipseKeys.projectFlavor := EclipseProjectFlavor.Java,
+  //  EclipseKeys.projectFlavor := EclipseProjectFlavor.ScalaIDE,
+
   EclipseKeys.withBundledScalaContainers := false,
   EclipseKeys.createSrc := EclipseCreateSrc.Default + EclipseCreateSrc.Resource,
   EclipseKeys.eclipseOutput := Some(".target"),
   EclipseKeys.withSource := true,
-  EclipseKeys.withJavadoc := true,
+  EclipseKeys.withJavadoc := true
+
   // avoid some scala specific source directories
-  unmanagedSourceDirectories in Compile := Seq((javaSource in Compile).value),
-  unmanagedSourceDirectories in Test := Seq((javaSource in Test).value)
+  // SCALA SUPPORT 2: Comment the following two lines.
+  ,  unmanagedSourceDirectories in Compile := Seq((javaSource in Compile).value)
+  ,  unmanagedSourceDirectories in Test := Seq((javaSource in Test).value)
```

Comment:
To remove the unusual comma handling and the need for users to search for multiple "SCALA SUPPORT" comments, you could move these keys so that they are directly below the `projectFlavor` part.